src/drivers/amd/rembrandt/acp_sw_audio_dma.c

```diff
@@ -0,0 +1,640 @@
+// SPDX-License-Identifier: BSD-3-Clause//
+//Copyright(c) 2023 AMD. All rights reserved.
+//
+//Author:       Basavaraj Hiregoudar <basavaraj.hiregoudar@amd.com>
+//              Maruthi Machani <maruthi.machani@amd.com>
+
+#include <rtos/atomic.h>
+#include <sof/audio/component.h>
+#include <rtos/bit.h>
+#include <sof/drivers/acp_dai_dma.h>
+#include <rtos/interrupt.h>
+#include <rtos/timer.h>
+#include <rtos/alloc.h>
+#include <rtos/clk.h>
+#include <sof/lib/cpu.h>
+#include <sof/lib/dma.h>
+#include <sof/lib/io.h>
+#include <sof/lib/pm_runtime.h>
+#include <sof/lib/notifier.h>
+#include <sof/platform.h>
+#include <sof/schedule/schedule.h>
+#include <rtos/spinlock.h>
+#include <sof/math/numbers.h>
+#include <sof/trace/trace.h>
+#include <ipc/topology.h>
+#include <user/trace.h>
+#include <errno.h>
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <platform/fw_scratch_mem.h>
+#include <platform/chip_registers.h>
+
+/*b414df09-9e31-4c59-8657-7afc8deba70c*/
+DECLARE_SOF_UUID("acp-sw-audio", acp_sw_audio_uuid, 0xb414df09, 0x9e31, 0x4c59,
+		 0x86, 0x57, 0x7a, 0xfc, 0x8d, 0xeb, 0xa7, 0x0c);
+DECLARE_TR_CTX(acp_sw_audio_tr, SOF_UUID(acp_sw_audio_uuid), LOG_LEVEL_INFO);
+
+#define SW0_AUDIO_FIFO_SIZE			128
+#define SW0_AUDIO_TX_FIFO_ADDR		0
+#define SW0_AUDIO_RX_FIFO_ADDR		(SW0_AUDIO_TX_FIFO_ADDR + SW0_AUDIO_FIFO_SIZE)
+
+#define SW0_BT_FIFO_SIZE		128
+#define SW0_BT_TX_FIFO_ADDR		(SW0_AUDIO_RX_FIFO_ADDR + SW0_AUDIO_FIFO_SIZE)
+#define SW0_BT_RX_FIFO_ADDR		(SW0_BT_TX_FIFO_ADDR + SW0_BT_FIFO_SIZE)
+
+#define SW0_HS_FIFO_SIZE		128
+#define SW0_HS_TX_FIFO_ADDR		(SW0_BT_RX_FIFO_ADDR + SW0_BT_FIFO_SIZE)
+#define SW0_HS_RX_FIFO_ADDR		(SW0_HS_TX_FIFO_ADDR + SW0_HS_FIFO_SIZE)
+
+#define SW1_FIFO_SIZE			128
+#define SW1_TX_FIFO_ADDR		(SW0_HS_RX_FIFO_ADDR + SW1_FIFO_SIZE)
+#define SW1_RX_FIFO_ADDR		(SW1_TX_FIFO_ADDR + SW1_FIFO_SIZE)
```

Comment:
You probably want a comment saying that there are two controllers with dedicated FIFOs/ports for audio, BT, etc.
I know this part because I've seen the Linux patches but other reviewers did not.

Reply:
done